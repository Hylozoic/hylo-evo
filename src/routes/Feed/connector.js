import { connect } from 'react-redux'
import { SAMPLE_FEED_ITEMS } from './sampleData'

function mapStateToProps (state) {
  return {
    feedItems: SAMPLE_FEED_ITEMS
  }
}

function mapDispatchToProps (dispatch, props) {
  return {}
}

export default connect(mapStateToProps, mapDispatchToProps)
