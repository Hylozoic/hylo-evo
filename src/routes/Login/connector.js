import { connect } from 'react-redux'
import { checkLogin, login } from './actions'
export default connect(null, {checkLogin, login})
