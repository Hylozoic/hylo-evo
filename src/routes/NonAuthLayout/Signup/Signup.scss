$third-party-icon-size: 14px;

.background {
  width: 100vw;
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}


.particlesBackgroundWrapper {
  position: absolute;
  display: block;
  top: 0;
  left: 0;
  z-index: -1000;
  height: 100vh;
  width: 100vw;
  line-height: 100vh;
}

.container {
  background-color: white;
  padding: 50px 40px 40px;
  border-radius: 8px;
  width: 420px;
}

.title {
  composes: hdr-headline from 'css/typography.scss';
  color: $color-caribbean-green;
  text-align: center;
}

.blurb {
  composes: bdy-lt-sm from 'css/typography.scss';
  text-align: center;
  margin-bottom: 30px;
}

.field {
  display: block;
  margin-bottom: 20px;
}

.field-label {
  composes: caption-drk-lg from 'css/typography.scss';
  display: block;
}

.submit {
  width: 100%;
  text-align: center;
  cursor: pointer;
}

.connect-label {
  composes: caption-lt-lg from 'css/typography.scss';
  margin-top: 40px;
  margin-bottom: 20px;
  text-align: center;
}

.auth-buttons {
  composes: caption-lt-lg from 'css/typography.scss';
  background-color: $color-seashell;
  border-radius: 42px;
  display: flex;
  justify-content: center;
  align-items: center;
  > a {
    display: block;
    text-align: center;
    height: 42px;
    line-height: 40px;
    vertical-align: middle;
    flex: 1;
    cursor: pointer;
  }
}

.auth-icon {
  margin-right: 11px;
  font-size: $third-party-icon-size;
}

.facebook {
  span {
    position: relative;
    top: 1px;
  }
  color: inherit;
  text-decoration: none !important;
}

.facebook:hover, .facebook:focus {
  color: inherit;
}

.google {
  border-left: 2px solid white;
  border-right: 2px solid white;
  span {
    position: relative;
    top: 1px;
  }
  color: inherit;
  text-decoration: none !important;
}

.google:hover, .google:focus {
  color: inherit;
}

.linkedin {

}

.error {
  color: $color-amaranth;
  background-color: transparentize($color-amaranth, 0.9);
  border: 1px solid $color-amaranth;
  border-radius: 4px;
  padding: 9px 16px;
  margin-bottom: 20px;
}

.below-container {
  margin-top: $space-2x;
  a {
    color: white !important;
    cursor: pointer;
  }
}
