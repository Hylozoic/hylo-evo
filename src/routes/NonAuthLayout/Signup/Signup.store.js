import { SIGNUP } from 'store/constants'

export function signup (name, email, password) {
  return {
    type: SIGNUP,
    payload: {
      api: {
        method: 'post',
        path: '/noo/user',
        params: {
          name,
          email,
          password,
          login: true
        }
      }
    }
  }
}
