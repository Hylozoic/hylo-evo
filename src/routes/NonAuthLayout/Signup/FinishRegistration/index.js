import { withCookies } from 'react-cookie'

import component from './FinishRegistration'
import connector from './FinishRegistration.connector'

export default withCookies(connector(component))
