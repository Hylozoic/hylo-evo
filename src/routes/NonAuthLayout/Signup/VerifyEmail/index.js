import component from './VerifyEmail'
import connector from './VerifyEmail.connector'
export default connector(component)
