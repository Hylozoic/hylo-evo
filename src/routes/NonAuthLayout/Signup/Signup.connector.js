import { connect } from 'react-redux'
import getLoginError from 'store/selectors/getLoginError'
import { getReturnToURL, resetReturnToURL } from 'router/AuthRoute/AuthRoute.store'
import mobileRedirect from 'utils/mobileRedirect'
import { signup } from './Signup.store'

export function mapStateToProps (state) {
  return {
    error: getLoginError(state),
    returnToURL: getReturnToURL(state),
    downloadAppUrl: mobileRedirect()
  }
}

export const mapDispatchToProps = {
  signup,
  resetReturnToURL
}

export function mergeProps (stateProps, dispatchProps, ownProps) {
  return {
    ...stateProps,
    ...dispatchProps,
    ...ownProps,
    redirectOnSignIn: (defaultPath) => {
      dispatchProps.resetReturnToURL()
      dispatchProps.push(stateProps.returnToURL || defaultPath)
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps, mergeProps)
