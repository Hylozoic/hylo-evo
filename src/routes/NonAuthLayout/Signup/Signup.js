import React from 'react'
import TextInput from 'components/TextInput'
import Button from 'components/Button'
import Icon from 'components/Icon'
import { formatError } from '../util'
import DownloadAppModal from 'components/DownloadAppModal'
import './Signup.scss'

export default class Signup extends React.Component {
  constructor (props) {
    super(props)
    this.state = {}
  }

  submit = () => {
    this.props.signup(this.state.name, this.state.email, this.state.password)
  }

  render () {
    const { className, downloadAppUrl } = this.props
    const setState = key => event => this.setState({[key]: event.target.value})
    return <div className={className}>
      {downloadAppUrl && <DownloadAppModal url={downloadAppUrl} />}
      <h1 styleName='title'>Welcome to Hylo</h1>
      <p styleName='blurb'>Stay connected, organized, and engaged with your community.</p>
      {this.props.error && formatError(this.props.error, 'Signup')}
      <div styleName='field'>
        <label styleName='field-label'>Full name</label>
        <TextInput type='text' name='name' onChange={setState('name')}
          inputRef={input => { this.name = input }} autoFocus />
      </div>
      <div styleName='field'>
        <label styleName='field-label'>Your email address</label>
        <TextInput type='text' name='email' onChange={setState('email')}
          inputRef={input => { this.email = input }} />
      </div>
      <div styleName='field'>
        <label styleName='field-label'>Password</label>
        <TextInput type='password' name='password'
          onChange={setState('password')}
          onEnter={this.submit} />
      </div>
      <Button styleName='submit' label='Sign Up' onClick={this.submit} />
      <p styleName='connect-label'>Or connect with:</p>
      <div styleName='auth-buttons'>
        <a styleName='facebook' onClick={this.submit}>
          <Icon name='Facebook' styleName='auth-icon' />
          Facebook
        </a>
        <a styleName='google' onClick={this.submit}>
          <Icon name='Google' styleName='auth-icon' />
          Google
        </a>
      </div>
    </div>
  }
}
