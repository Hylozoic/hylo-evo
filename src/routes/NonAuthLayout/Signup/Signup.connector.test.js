import { mapDispatchToProps } from './Signup.connector'

describe('Signup', () => {
  it('should call signup', () => {
    expect(mapDispatchToProps.signup('name', 'test@hylo.com', 'testPassword')).toMatchSnapshot()
  })
})
