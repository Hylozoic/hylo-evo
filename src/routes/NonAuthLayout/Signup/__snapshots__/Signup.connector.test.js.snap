// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup should call signup 1`] = `
Object {
  "payload": Object {
    "api": Object {
      "method": "post",
      "params": Object {
        "email": "test@hylo.com",
        "login": true,
        "name": "name",
        "password": "testPassword",
      },
      "path": "/noo/user",
    },
  },
  "type": "SIGNUP",
}
`;
